src/cascadia/ShellExtension/OpenTerminalHere.cpp

```diff
@@ -106,10 +106,25 @@ HRESULT OpenTerminalHere::GetState(IShellItemArray* /*psiItemArray*/,
     // E_PENDING and this object will be called back on a background thread with
     // fOkToBeSlow == TRUE
 
-    // We however don't need to bother with any of that, so we'll just return
-    // ECS_ENABLED.
+    // We however don't need to bother with any of that.
+
+    // If no item was selected when the context menu was opened and Explorer
+    // is not at a valid path (e.g. This PC or Quick Access), we should hide
+    // the verb from the context menu.
+    if (psiItemArray == nullptr)
+    {
+        const auto path = this->_GetPathFromExplorer();
+        *pCmdState = path.empty() ? ECS_HIDDEN : ECS_ENABLED;
+    }

```

Comment:
Can `psiItemArray` be `nullptr`? If it's usually never `nullptr`, we should probably just return `ECS_ENABLED`. This would at least not be worse than the previous solution.

And I believe `_GetPathFromExplorer` might be too slow for `fOkToBeSlow == FALSE`. 🤔

Reply:
@lhecker there's also an issue in older versions of Windows where `psiItemArray` would be `nullptr` even when the user had selected an item.